src/localsocketcontroller.cpp

```diff
@@ -107,18 +107,8 @@ void LocalSocketController::initializeInternal() {
   }
   m_daemonState = eInitializing;
 
-#ifdef MZ_WINDOWS
-  QString path = "\\\\.\\pipe\\mozillavpn";
-#else
-  QString path = "/var/run/mozillavpn/daemon.socket";
-  if (!QFileInfo::exists(path)) {
-    path = "/tmp/mozillavpn.socket";
-  }
-#endif
-
-  logger.debug() << "Connecting to:" << path;
-  m_socket->abort();
```

Comment:
I think I missed the removal of `m_socket->abort()` good catch! That is there for the daemon recovery protocol so that we drop any unsent data and reset the socket before trying to re-connect to the Daemon. We probably need to put that back.

Reply:
As for why the `path` stuff got removed it has become the job of the caller to provide the path to the `LocalSocketController` constructor. This code basically just got moved out and became `Controller::useLocalSocketPath()` instead.

The reasoning here is that we might need to override that path at runtime if we are connecting to a different daemon for testing purposes. When we run the client with the `--testing` option we create a mocked daemon that generates an ephemeral path and connect to it instead of the real daemon. This allows the tests to be run without needing to install the real daemon, or needing permissions to create paths in `/var`.